g3/lowpan6_common.c

```diff
@@ -755,8 +832,7 @@ lowpan6_decompress_hdr(u8_t *lowpan6_buffer, size_t lowpan6_bufsize,
         return ERR_VAL;
       }
 #if LWIP_6LOWPAN_NUM_CONTEXTS > 0
-      ip6hdr->src.addr[0] = lowpan6_contexts[i].addr[0];
-      ip6hdr->src.addr[1] = lowpan6_contexts[i].addr[1];
+      lowpan6_context_set(&lowpan6_contexts[i], ip6hdr->src.addr);
```

Comment:
possible unaligned access - the `src` field is packed (it's of type `ip6_addr_p_t`) so the compiler knows the `addr` may not be 4-byte aligned. Here we're casting to u32_t so that's undefined behaviour.

Overall note: protocol parsing is hard because of triggering unaligned access - the basic rules are:
 - cast from u8* buffer only to packed structs
 - move data to aligned structs either by memcpy or operating byte-by-byte

I probably won't find every unaligned access in this code, You may try to run Your test programs with UBsan/ASan enabled, so some of them can be caught automatically